#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_genisoimage.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use genisoimage instead of mkisofs, which is deprecated

@DPATCH@
diff -urNad vdr-plugin-burn-0.1.0~pre21~/vdrburn-archive.sh vdr-plugin-burn-0.1.0~pre21/vdrburn-archive.sh
--- vdr-plugin-burn-0.1.0~pre21~/vdrburn-archive.sh	2006-08-27 20:20:13.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/vdrburn-archive.sh	2008-05-17 21:29:48.000000000 +0200
@@ -12,7 +12,7 @@
 
 case $1 in
 	mkiso)
-		mkisofs -V "$DISC_ID" -f -r -J -l -allow-lowercase -allow-multidot \
+		genisoimage -V "$DISC_ID" -f -r -J -l -allow-lowercase -allow-multidot \
 				-graft-points $GRAFT_POINTS >"$ISO_FILE"
 	;;
 
@@ -27,7 +27,7 @@
 	;;
 
 	pipeiso)
-		mkisofs -V "$DISC_ID" -f -r -J -l -allow-lowercase -allow-multidot \
+		genisoimage -V "$DISC_ID" -f -r -J -l -allow-lowercase -allow-multidot \
 				-graft-points $GRAFT_POINTS | tee "$ISO_FILE" > $ISO_PIPE
 	;;
 
diff -urNad vdr-plugin-burn-0.1.0~pre21~/vdrburn-dvd.sh vdr-plugin-burn-0.1.0~pre21/vdrburn-dvd.sh
--- vdr-plugin-burn-0.1.0~pre21~/vdrburn-dvd.sh	2006-08-27 20:34:30.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/vdrburn-dvd.sh	2008-05-17 21:29:32.000000000 +0200
@@ -207,7 +207,7 @@
 	;;
 
 	mkiso)
-		mkisofs -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" > "$ISO_FILE"
+		genisoimage -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" > "$ISO_FILE"
 	;;
 
 	burndir)
@@ -224,12 +224,12 @@
         if [ $BURN_SPEED -gt 0 ]; then
         	SPEED="speed=$(($BURN_SPEED * 4))"
         fi
-        mkisofs -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" \
+        genisoimage -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" \
 			| cdrecord "dev=$DVD_DEVICE" driveropts=burnfree fs=10m $SPEED -
     ;;
 
 	pipeiso)
-		mkisofs -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" \
+		genisoimage -V "$DISC_ID" -dvd-video "$DVDAUTHOR_PATH" \
 			| tee "$ISO_FILE" > "$ISO_PIPE"
 	;;
 
